src/app/service.rs

```diff
@@ -95,7 +95,7 @@ impl<T: Send + Sync> Service<http::Request<Incoming>> for AppService<T> {
         // segment.
         for (route, param) in self.app.router.traverse(parts.uri.path()) {
             // Add the matched route's middleware to the call stack.
-            next.extend(route.cloned());
```

Comment:
A bit of scope creep but I'm just undoing what I did in #239. I changed my mind and decided it's better to be explicit about this. It's nice to see that the cost of adding an additional middleware to a route is an atomic increment.